coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -529,7 +530,19 @@ private SqlNode visitFunctionInternal(ASTNode node, ParseContext ctx, SqlLiteral
   protected SqlNode visitSelectExpr(ASTNode node, ParseContext ctx) {
     List<SqlNode> sqlNodes = visitChildren(node, ctx);
     if (sqlNodes.size() == 1) {
-      return sqlNodes.get(0);
+      final SqlNode sqlNode = sqlNodes.get(0);
+      // if select a field from a struct column, like
+      // SELECT table.struct_column.field_name FROM table
+      // we need to add an alias to avoid null pointer exception:
+      // SELECT table.struct_column.field_name AS field_name FROM table
+      if (sqlNode.getKind() == SqlKind.IDENTIFIER) {
+        List<String> names = ((SqlIdentifier) sqlNode).names;
+        final int size = names.size();
+        if (size > 2) {
```

Comment:
a) what is the deal with  `table.struct_column.*`, does this need an alias as well? 
b) is there a better/more accurate way to identify struct columns? What if we have `user.table.non_struct_col` or `struct_column.field` ?  (we would need to add these cases to unit tests)